repo2docker/buildpacks/pipfile/__init__.py

```diff
@@ -0,0 +1,133 @@
+"""Generates Dockerfiles based on an input matrix based on Python."""
+import os
+import re
+
+from ..conda import CondaBuildPack
+
+
+class PipfileBuildPack(CondaBuildPack):
+    """Setup Python with pipfile for use with a repository."""
+
+    @property
+    def python_version(self):
+        """
+        Detect the Python version declared in a `Pipfile.lock`, `Pipfile`, or
+        `runtime.txt`. Will return 'x.y' if version is found (e.g '3.6'), or a
+        Falsy empty string '' if not found.
+        """
+
+        if hasattr(self, "_python_version"):
+            return self._python_version
+
+        files_to_search_in_order = [
+            {
+                "path": self.binder_path("Pipfile.lock"),
+                "pattern": r"\s*\"python_(?:full_)?version\": \"?([0-9a-z\.]*)\"?",  # '            "python_version": "3.6"'
```

Comment:
You could use `'` instead of `"` for the string itself to avoid having to escape the `"` inside the regex pattern.

Is it Ok to only match `"python_version": "3.6"` and not `"python_version": '3.6'` or other quote combos?

Reply:
From the experiment in https://github.com/jupyter/repo2docker/pull/649#issuecomment-504667325 I think we need to handle both kinds of quotes. Which makes me think of the old regex saying: I had a problem that I tried to solve with a regex, now I have two problems :)

Reply:
lol on the quote :D I'll look into it!